lib/internal/loader/Loader.js

```diff
@@ -71,8 +41,8 @@ class Loader {
     this._dynamicInstantiate = undefined;
   }
 
-  async resolve(specifier, parentURL = this.base) {
-    if (typeof parentURL !== 'string')
+  async resolve(specifier, parentURL) {
+    if (parentURL !== undefined && typeof parentURL !== 'string')
```

Comment:
I'm not sure I like the new logic (of using `parentURL !== undefined` instead of an explicit check) - If possible I'd at least make the check a getter or a function and call that instead.

Reply:
Loaders exist in their own context, without even a `this` reference currently, so the alternative channel would be arguments. So the idea of `parentURL === undefined` is that it avoids the need for a breaking change in the arguments when we already have it, and it makes logical API sense in terms of the tree structure of modules.

Reply:
Right, all I'm saying is that I'd likely put a function called `isMain` and pass `parentURL` to it to make it explicit - without the context of this PR it wouldn't be immediately obvious to me what it does

Reply:
We might want to be a bit careful about over abstracting here if we ever want to support a `--require` like flag for ESM, which I presume also would not have a `parentURL`, but wouldn't be a "main"?

Reply:
Sure,  we could assign to `isMain` first to make this clearer.